File: app/src/main/java/com/kode/utair/data/repository/datasource/WeatherPreferences.java
package com.kode.utair.data.repository.datasource;

import android.content.Context;
import android.content.SharedPreferences;

import com.kode.utair.domain.models.WeatherSettings;

import javax.inject.Inject;

public class WeatherPreferences {

    private static final String APP_PREFS_FILE_NAME = "app_preferences";

    private static final String DEPART_CITY_KEY = "depart_city_key";
    private static final String ARRIVE_CITY_KEY = "arrive_city_key";


    private SharedPreferences appPreferences;

    @Inject
    public WeatherPreferences(Context context) {
        this.appPreferences = context.getSharedPreferences(APP_PREFS_FILE_NAME, Context.MODE_PRIVATE);
    }

    public WeatherSettings getWeatherSettings() {
        String departCity = appPreferences.getString(DEPART_CITY_KEY, null);
        String arriveCity = appPreferences.getString(ARRIVE_CITY_KEY, null);
        return new WeatherSettings(departCity, arriveCity);
    }


    public void saveWeatherSettings(WeatherSettings weatherSettings) {
        if (weatherSettings.getDepartCity() != null) {
            appPreferences.edit().putString(DEPART_CITY_KEY, weatherSettings.getDepartCity()).apply();
        }

        if (weatherSettings.getArriveCity() != null) {
            appPreferences.edit().putString(ARRIVE_CITY_KEY, weatherSettings.getArriveCity()).apply();
        }
    }


}

File: app/src/main/java/com/kode/utair/data/repository/datasource/WeatherPreferences.kt
package com.kode.utair.data.repository.datasource

import android.content.Context
import android.content.SharedPreferences
import com.kode.utair.domain.models.WeatherSettings
import javax.inject.Inject

class WeatherPreferences @Inject constructor(context: Context) {

    private val appPreferences: SharedPreferences

    fun getWeatherSettings(): WeatherSettings {
        val departCity = appPreferences.getString(DEPART_CITY_KEY, "")!!
        val arriveCity = appPreferences.getString(ARRIVE_CITY_KEY, "")!!
        return WeatherSettings(departCity, arriveCity)
    }

    fun saveWeatherSettings(weatherSettings: WeatherSettings) {
        if (weatherSettings.departCity != null) {
            appPreferences.edit().putString(DEPART_CITY_KEY, weatherSettings.departCity).apply()
        }
        if (weatherSettings.arriveCity != null) {
            appPreferences.edit().putString(ARRIVE_CITY_KEY, weatherSettings.arriveCity).apply()
        }
    }

    companion object {
        private const val APP_PREFS_FILE_NAME = "app_preferences"
        private const val DEPART_CITY_KEY = "depart_city_key"
        private const val ARRIVE_CITY_KEY = "arrive_city_key"
    }

    init {
        appPreferences = context.getSharedPreferences(APP_PREFS_FILE_NAME, Context.MODE_PRIVATE)
    }
}